packages/rol/adapters/thyra/src/function/ROL_ThyraProductME_Objective.hpp

```diff
@@ -79,9 +79,9 @@ class ThyraProductME_Objective : public Objective<Real> {
     Thyra::ModelEvaluatorBase::InArgs<Real> inArgs = thyra_model.createInArgs();
     Thyra::ModelEvaluatorBase::OutArgs<Real> outArgs = thyra_model.createOutArgs();
 
-
     outArgs.set_g(g_index, g);
-    for(int i=0; i<p_indices.size(); ++i)
+    const int theSize = (int) p_indices.size();
```

Comment:
@MicheldeMessieres, can you please use `static_cast<int>(...)` instead of a C-style cast?  Change this to:

```
const int theSize = static_cast<int>(p_indices.size());
```

C-style casts are to be avoided at all costs in C++ code (they do evil things).  See Item 27 "Minimize casting" in "Effective C++ Third Edition" by Scott Meyers.